File: programs/linked_list.py
class Node:
    def __init__(self,val,next = None):
        self.val = val
        self.next = next

class Linked_list:
    def __init__(self):
        self.head = None

    def add_node(self,val):
        new_node = Node(val)
        if (not self.head):
            self.head = new_node
            return
        current = self.head
        while(current.next):
            current = current.next
        current.next = new_node
    
    def print_ll(self):
        current = self.head
        while(current):
            print(current.val,end = (" "))
            current = current.next
        print()

    def search_ll(self,t):
        if (not self.head):
            return -1
        current = self.head
        while(current):
            if (current.val == t):
                return current
            current = current.next
        return -1

l1 = Linked_list()
l1.add_node(10)
l1.add_node(11)
l1.add_node(12)
l1.add_node(13)
l1.add_node(14)
l1.add_node(15)
l1.add_node(16)
l1.add_node(17)
l1.add_node(18)
l1.print_ll()
l2 = Linked_list()
pos = l2.search_ll(8)
print(pos)

File: programs/linked_list.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

struct node{
    int data;
    struct node* next = NULL;
};

// struct linked_list{
//     struct node *head;
// };
void print_ll(struct node* head) {
    while(head) {
        cout << head -> data << " ";
        head = head -> next;
    }
    cout << endl;
}

int main() {
    struct node *head = new node();
    struct node *tail; tail = head;
    struct node *p, *prev;
    head->data = 10; head->next = NULL; prev = head;
    for (int i = 1 ; i < 30 ; i *= 3) {
        p = new node();
        p -> data = i; p->next = NULL;
        prev -> next = p;
        prev = p;
    }
    struct node *head2 = new node();
    head2 -> data = 30; prev = head2; head2->next = NULL;
    for (int i = 2 ; i < 30 ; i *= 3) {
        p = new node();
        p -> data = i; p->next = NULL;
        prev -> next = p;
        prev = p;
    }
    print_ll(head); print_ll(head2);
    return 0;
}